components/query-controls/category-select.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * WordPress dependencies
+ */
+import { buildTermsTree } from '@wordpress/utils';
+
+/**
+ * Internal dependencies
+ */
+import TreeSelect from '../tree-select';
+
+export default function CategorySelect( { label, noOptionLabel, categories, selectedCategory, onChange } ) {
+	const termsTree = buildTermsTree( categories );
```

Comment:
Just wanted to leave a note that it is suboptimal to compute this on every prop change. Whenever `selectedCategory` changes this will also need to be processed. @aduth, does it make sense to use memoize here to have this logic executed only when the list of `categories` changes?

Reply:
I think it is a really nice suggestion maybe we can simply add memoize to buildTermsTree function.